userspace/libsinsp/filterchecks.cpp

```diff
@@ -5923,6 +5925,12 @@ uint8_t* sinsp_filter_check_event::extract(sinsp_evt *evt, OUT uint32_t* len, bo
 					m_u32val = 1;
 				}
 
+				if(m_field_id == TYPE_ISOPEN_CREATE &&
```

Comment:
> BTW what do we want by this filtercheck? do we need to involve also the CREAT syscall and the O_TMPFILE flag? @incertum @terylt

In response to @Andreagit97 questions, yes I would vote for this filtercheck to be true for either scenario, that is, the current implementation plus the O_TMPFILE scenario, but I haven't looked into the details on how it should be.

Reply:
Ok in this case i would probably use a dedicated `case` for `TYPE_ISOPEN_CREATE` and something like that:
```c
case TYPE_ISOPEN_CREATE:
{
	uint16_t etype = evt->get_type();
	m_u32val = 0;
	sinsp_evt_param *parinfo = nullptr;

	if(etype == PPME_SYSCALL_OPEN_X ||
			 etype == PPME_SYSCALL_OPENAT_E ||
			 etype == PPME_SYSCALL_OPENAT_2_X ||
			 etype == PPME_SYSCALL_OPENAT2_X ||
			 etype == PPME_SYSCALL_OPEN_BY_HANDLE_AT_X)
	{
		bool is_new_version = etype == PPME_SYSCALL_OPENAT_2_X || etype == PPME_SYSCALL_OPENAT2_X;
		parinfo = evt->get_param(is_new_version ? 3 : 2);
		ASSERT(parinfo->m_len == sizeof(uint32_t));
		uint32_t flags = *(uint32_t *)parinfo->m_val;

		// If the flag is set we are use the file is created
		if(flags & PPM_O_F_CREATED)
		{
			m_u32val = 1;
		}

		// If the flag is `PPM_O_TMPFILE` and the syscall is susseful the file is created
		if(flags & PPM_O_TMPFILE)
		{
			parinfo = evt->get_param(0);
			ASSERT(parinfo->m_len == sizeof(int64_t));
			int64_t retval = *(int64_t *)parinfo->m_val;
			if(retval >= 0)
			{
				m_u32val = 1;
			}
		}
	}
	
	RETURN_EXTRACT_VAR(m_u32val);
}
break;	
```

WDYT?

Reply:
cc @mrgian

Reply:
Yeah I also think this filtercheck should be true if `O_TMPFILE` is set.
I agree with @Andreagit97 implementation
We don't need to do anything special for this one, we just need to check the return value of the syscall to know if the file was actually created, right?

Reply:
Thanks everyone for adding this and thanks @Andreagit97 for helping :)